packages/legend-query-builder/src/components/shared/BasicValueSpecificationEditor.tsx

```diff
@@ -196,7 +196,10 @@ const StringPrimitiveInstanceValueEditor = observer(
     } = props;
     const useSelector = Boolean(selectorConfig);
     const applicationStore = useApplicationStore();
-    const value = valueSpecification.values[0] as string;
+    const value =
```

Comment:
when we do see the case where this returns `undefined` (i.e. has less than 1 item)? if we're just trying to get around TS here, we can use `guaranteeIsString(...)`

Reply:
i think we should go back to what it was before. looking at the code we added this because we weren't populating the watermark string value.